pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -101,7 +101,7 @@
         category = CATEGORY_SERVER,
         doc = "The port for serving binary protobuf requests"
     )
-    private Integer brokerServicePort = 6650;
+    private Integer brokerServicePort;
```

Comment:
Wouldn't this disable the "clear-text" service by default?

Reply:
> Wouldn't this disable the "clear-text" service by default?

yes. with this change, "clear-text" will disable the service.

Reply:
I mean, will it be disabled by default?

Reply:
> Wouldn't this disable the "clear-text" service by default?

yes. in order to disable it, right now, we are considering that user will give "clear-text". So, by default if user doesn't provide port then it will be disable. also [pulsar-service fails](https://github.com/apache/pulsar/blob/be20d32406f0f72d8cb21ce964b76e65bf604695/pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java#L342) if user doesn't provide both tls and non-tls ports.

However, I also agree that broker service should start with default port if none of the port is provided by default. so, we can have multiple options to disable service on non-tls port
1. user can provide clear-text that will disable it by default.
2. user can explicitly provide -1 value  to disable non-tls port else by default it will listen on non-tls ports
3. if both tls and non-tls ports are not provided then broker will start service on default non-tls ports.

I think we can do 2nd option to explicitly disable non-tls ports. any thoughts?

Reply:
what if user provides `brokerServicePort=` in `broker.conf` instead of leaving that line empty?

Reply:
then right now, [as we can see in this test](https://github.com/apache/pulsar/blob/be20d32406f0f72d8cb21ce964b76e65bf604695/pulsar-broker-common/src/test/java/org/apache/pulsar/common/configuration/PulsarConfigurationLoaderTest.java#L103), `PulsarConfigurationLoader` doesn't update the field and broker considers default field value so, it will be always 8080 if we keep `private Integer brokerServicePort = 6650;`.  let me check if we can make change at `PulsarConfigurationLoader` where it sets field value null if not value is provided in config file.